rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -0,0 +1,484 @@
+    ///
+    /// Returns the updated [`AssetEndpointProfile`] once updated.
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`InvalidArgument`](ErrorKind::InvalidArgument)
+    /// if the `timeout` is zero or > `u32::max`, or there is an error building the request.
+    ///
+    /// [`struct@Error`] of kind [`SerializationError`](ErrorKind::SerializationError)
+    /// if there is an error serializing the request.
+    ///
+    /// [`struct@Error`] of kind [`ServiceError`](ErrorKind::ServiceError)
+    /// if there is an error returned by the ADR Service.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
+    pub async fn update_asset_endprofile_status(
+        &self,
+        source: UpdateAssetEndpointProfileStatusReq,
+        timeout: Duration,
+    ) -> Result<AssetEndpointProfile, AzureDeviceRegistryError> {
+        let command_request = rpc_command::invoker::RequestBuilder::default()
+            .payload(source.into())
+            .map_err(|e| AzureDeviceRegistryError(ErrorKind::SerializationError(e.to_string())))?
+            .timeout(timeout)
+            .build()
+            .map_err(|e| AzureDeviceRegistryError(ErrorKind::InvalidArgument(e.to_string())))?;
+
+        let result = self
+            .update_asset_endpoint_profile_status_command_invoker
+            .invoke(command_request)
+            .await;
+
+        match result {
+            Ok(response) => Ok(response.payload.updated_asset_endpoint_profile),
+            Err(e) => Err(AzureDeviceRegistryError(ErrorKind::from(
+                ErrorKind::AIOProtocolError(e),
+            ))),
+        }
+    }
+
+    /// Notifies the Azure Device Registry service that client is listening for asset updates.
+    ///
+    /// # Arguments
+    /// * `notification_type` - The type of notification to send, true for "On" and false for "Off".
+    /// * `timeout` - The duration until the Client stops waiting for a response to the request, it is rounded up to the nearest second.
+    ///
+    /// Returns the [`NotificationResponse`].
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`InvalidArgument`](ErrorKind::InvalidArgument)
+    /// if the `timeout` is zero or > `u32::max`, or there is an error building the request.
+    ///
+    /// [`struct@Error`] of kind [`SerializationError`](ErrorKind::SerializationError)
+    /// if there is an error serializing the request.
+    ///
+    /// [`struct@Error`] of kind [`ServiceError`](ErrorKind::ServiceError)
+    /// if there is an error returned by the ADR Service.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
+    pub async fn notify_asset_endpoint_profile_update(
```

Comment:
let's match this wording with the state store client and split this into two functions:
observe_asset_endpoint_profile_updates() and unobserve_asset_endpoint_profile_updates()